### PR TITLE
completed assignment 4

### PR DESCRIPTION
**Pull request by sumitnegikv369**
Screenshot from 2024-04-09 09-32-45

### CODE REVIEW

**Comment by sumitnegikv369** [comment]
This is the last week of my react training, requesting you to review my remaining PRs

File: myapp/src/day4/Question1/UI.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useState } from 'react'
+import './styles.css'
+
+const UI = () => {
+  const [value, setValue] = useState('')
+  return (
+    <div className="ques1">
+      <input
+        cvalue={value}
```
Comment (bhavya-goel) [inline myapp/src/day4/Question1/UI.jsx]: what does cvalue do?

File: myapp/src/day4/Question6/Todo.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <div className="ques6">
+      <div className="inputarea">
+        <input
+          type="text"
+          placeholder="add a new task"
+          value={value}
+          onChange={(e) => setValue(e.target.value)}
+        />
+        <button onClick={handleTask}>Add</button>
```
Comment (bhavya-goel) [inline myapp/src/day4/Question6/Todo.jsx:25]: check for empty values before adding to list

File: myapp/src/day4/Question7/SearchFilter.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState } from 'react'
+import './styles.css'
+
+const SearchFilter = () => {
+  const [items, setItems] = useState([])
+  const [searchInput, setSearchInput] = useState('')
+  const filterSearch = async () => {
+    const res = await fetch('https://dummyjson.com/products')
```
Comment (bhavya-goel) [inline myapp/src/day4/Question7/SearchFilter.jsx]: get the data from API once and store it. simply filter values from it when user types anything. We don't need to hit API again

File: myapp/src/day4/Question10/Form.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        required
+        id="outlined-required"
+        label="Required"
+        type="password"
+        placeholder="Enter User Password"
+        autoComplete="current-password"
+        value={userInfo.password}
+        onChange={(e) => setUserInfo({ ...userInfo, password: e.target.value })}
+      />
+      <Button variant="contained">Send</Button>
```
Comment (bhavya-goel) [inline myapp/src/day4/Question10/Form.jsx]: no error messages ae displayed when required fields are not filled

**Review by bhavya-goel — CHANGES_REQUESTED** [review]

File: myapp/src/day4/Question1/UI.jsx (hunk 1 above)
Comment (sumitnegikv369) [reply myapp/src/day4/Question1/UI.jsx]: it was a typo, i had removed it now

File: myapp/src/day4/Question6/Todo.jsx (hunk 2 above)
Comment (sumitnegikv369) [reply myapp/src/day4/Question6/Todo.jsx:25]: okay, i had added if condition for that

File: myapp/src/day4/Question7/SearchFilter.jsx (hunk 3 above)
Comment (sumitnegikv369) [reply myapp/src/day4/Question7/SearchFilter.jsx]: done

**Review by bhavya-goel — APPROVED** [review]

**Pull request merged**